### PR TITLE
Adding Private DNS Swagger spec for API version 2018-09-01

### PR DESCRIPTION
**Pull request by muwaqar**
This PR was [previously approved](https://github.com/Azure/azure-rest-api-specs-pr/pull/580) against the private repository by both ARM and OpenAPI teams.

OpenAPI Hub was not used for this PR since it does not allow onboarding of a new resource type as of yet. This feedback has been given to Samer previously.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by sergey-shandar** [comment]
Waiting for codeowners approval.

File: specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json [hunk 1]
```diff
@@ -0,0 +1,1919 @@
...
+            },
+            "description": "Parameters supplied to the CreateOrUpdate operation."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfMatch",
+            "description": "The ETag of the Private DNS zone. Omit this value to always overwrite the current zone. Specify the last-seen ETag value to prevent accidentally overwritting any concurrent changes."
```
Comment (nschonni) [inline specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json]: overwritting -> overwriting

File: specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json [hunk 2]
```diff
@@ -0,0 +1,1919 @@
...
+            },
+            "description": "Parameters supplied to the Update operation."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfMatch",
+            "description": "The ETag of the Private DNS zone. Omit this value to always overwrite the current zone. Specify the last-seen ETag value to prevent accidentally overwritting any concurrent changes."
```
Comment (nschonni) [inline specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json]: overwritting -> overwriting

File: specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json [hunk 3]
```diff
@@ -0,0 +1,1919 @@
...
+            },
+            "description": "Parameters supplied to the CreateOrUpdate operation."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfMatch",
+            "description": "The ETag of the virtual network link to the Private DNS zone. Omit this value to always overwrite the current virtual network link. Specify the last-seen ETag value to prevent accidentally overwritting any concurrent changes."
```
Comment (nschonni) [inline specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json]: overwritting -> overwriting

File: specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json [hunk 4]
```diff
@@ -0,0 +1,1919 @@
...
+            },
+            "description": "Parameters supplied to the Update operation."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfMatch",
+            "description": "The ETag of the virtual network link to the Private DNS zone. Omit this value to always overwrite the current virtual network link. Specify the last-seen ETag value to prevent accidentally overwritting any concurrent changes."
```
Comment (nschonni) [inline specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json]: overwritting -> overwriting

File: specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json [hunk 5]
```diff
@@ -0,0 +1,1919 @@
...
+            },
+            "description": "Parameters supplied to the CreateOrUpdate operation."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfMatch",
+            "description": "The ETag of the record set. Omit this value to always overwrite the current record set. Specify the last-seen ETag value to prevent accidentally overwritting any concurrent changes."
```
Comment (nschonni) [inline specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json]: overwritting -> overwriting

File: specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json [hunk 6]
```diff
@@ -0,0 +1,1919 @@
...
+            },
+            "description": "Parameters supplied to the Update operation."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfMatch",
+            "description": "The ETag of the record set. Omit this value to always overwrite the current record set. Specify the last-seen ETag value to prevent accidentally overwritting concurrent changes."
```
Comment (nschonni) [inline specification/privatedns/resource-manager/Microsoft.Network/stable/2018-09-01/privatedns.json]: overwritting -> overwriting

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by sergey-shandar — APPROVED** [review]

**Comment by jianghaolu** [comment]
@AutorestCI regenerate azure-sdk-for-java

**Pull request merged**